scrapy/http/request/json_request.py

```diff
@@ -42,7 +44,7 @@ def __init__(self, *args, dumps_kwargs: Optional[dict] = None, **kwargs) -> None
     def dumps_kwargs(self) -> dict:
         return self._dumps_kwargs
 
-    def replace(self, *args, **kwargs) -> Request:
+    def replace(self, *args: Any, **kwargs: Any) -> Request:
```

Comment:
isn't it returning JsonRequest, not Request?

Reply:
Not if you pass `cls`.